sound/soc/sof/topology.c

```diff
@@ -3195,6 +3221,13 @@ static int sof_route_load(struct snd_soc_component *scomp, int index,
 		goto err;
 	}
 
+	if (source_swidget->id == snd_soc_dapm_buffer) {
```

Comment:
please comment what we are checing any why.

Reply:
done